de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/shipmentschedule/api/ShipmentScheduleEnqueuer.java

```diff
@@ -100,7 +100,8 @@ public Result createWorkpackages(
 			final int adPInstanceId,
 			final IQueryFilter<I_M_ShipmentSchedule> queryFilters,
 			final boolean useQtyPickedRecords,
-			final boolean completeShipments)
+			final boolean completeShipments,
```

Comment:
* suggestion: extract those params into one "parameter-pojo" or maybe extract just those params that are passed to the workpacakge..then you can extrac them back into the same pojo when you process the WP
  * by the way, think codebeat strongly agrees :-)
* is it OK for queryFilters to be null? please check it anannotate with `@Nullable` or `@NotNull` to make it clearer

Reply:
Note, The query Filters shall never be null. They shall, at least, contain the isActive condition. Annotating them with NonNull